### ISSUE TITLE
Cannot publish packages using `poetry publish` in interactive mode

### ISSUE THREAD

**Issue by MadJlzz** [python-poetry/poetry#5610]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: `Linux MadDows 4.19.104-microsoft-standard #1 SMP Wed Feb 19 06:37:35 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`
- **Poetry version**: `Poetry (version 1.2.0b1)`
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/MadJlzz/a8305c6df426928abff1e913c0aa8f50

## Issue

While working on a plugin for `poetry`, I was about to publish it when I encountered an error:
```shell
(.venv) madjlzz@MadDows:~/oss/poetry-artifactregistry-plugin$ poetry publish --build
Building poetry-artifactregistry-plugin (0.1.0)
  - Building sdist
  - Built poetry-artifactregistry-plugin-0.1.0.tar.gz
  - Building wheel
  - Built poetry_artifactregistry_plugin-0.1.0-py3-none-any.whl


  AttributeError

  'IO' object has no attribute 'ask'

  at .venv/lib/python3.9/site-packages/poetry/publishing/publisher.py:80 in publish
       76│         )
       77│         # Requesting missing credentials but only if there is not a client cert defined.
       78│         if not resolved_client_cert:
       79│             if username is None:
    →  80│                 username = self._io.ask("Username:")
       81│ 
       82│             # skip password input if no username is provided, assume unauthenticated
       83│             if username and password is None:
       84│                 password = self._io.ask_hidden("Password:")
```

`cleo` version is `1.0.0a4` and looking at the repository, I indeed could not find any `ask()` nor `ask_hidden()` method. It also seems that `ConsoleIO` does not exist anymore. It seems that the commit responsible of this is https://github.com/sdispater/cleo/commit/de55578da25c6b1736b8b818f21c1bacf7c2475d

See here: https://github.com/python-poetry/poetry/blob/feb11b133af2e94d6c5239decb4e614426f2a299/src/poetry/publishing/publisher.py#L79-L85

**Comment by MadJlzz** [comment]
Created also an issue in `cleo` as it breaks as well just after: https://github.com/sdispater/cleo/issues/134

**Comment by branchvincent** [comment]
Thanks for filing, this is a duplicate #4790 which should be fixed with #5085
